### PR TITLE
[VL][Core] SampleExec Operator Native Support 

### PR DESCRIPTION
**Pull request by gaoyangxiaozhu**
## What changes were proposed in this pull request?

The PR introduce the `SampleExec` native support (`SampleExecTransformer`) for velox backend, basically it transfer the `SampleExec` spark physical operator to velox `ProjectFilter` operator with `rand(seed) < fraction` expression.

The feature is under feature flag `spark.gluten.sql.columnarSampleEnabled` with default is false now.

SampleExec support two sample way
 1. Sampling with replacement (withReplacement is set to true)
 2. sampling without replacement  (default)

Right now, we only support for samling without replacement.

(Please fill in changes proposed in this fix)

(Fixes: \#5315)

## How was this patch tested?
locally test with turning on feature flag 

A case use `df.sample(0.5, 10)`

image

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by gaoyangxiaozhu** [comment]
@rui-mo / @zhouyuan / @zhli1142015  help review

File: gluten-core/src/main/scala/org/apache/gluten/execution/SampleExecTransformer.scala [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import org.apache.gluten.substrait.rel.{RelBuilder, RelNode}
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.catalyst.expressions.{Attribute, Expression, LessThan, Literal, Rand}
+import org.apache.spark.sql.execution.SparkPlan
+import org.apache.spark.sql.types.DoubleType
+
+import scala.collection.JavaConverters._
+
+case class SampleExecTransformer(
```
Comment (rui-mo) [inline gluten-core/src/main/scala/org/apache/gluten/execution/SampleExecTransformer.scala:35]: Would you add some description for how sample operator is translated into Substrait?

**Comment by gaoyangxiaozhu** [comment]
> > Right now, we only support for samling without replaceme
> 
> Will Sampling with replacement (withReplacement is set to true) fallback? how complex if we add support to Velox? how useful it is?

sample with relpacement allow pick up same row multiple times, that means it actually not simple filter but re-generating the data set,  that is not worked by simple convert to a filter rand expression. I don't yet figur out a good way to supprot that part unless to make a new `sampleNode` in Velox side.

File: gluten-core/src/main/scala/org/apache/gluten/execution/SampleExecTransformer.scala (hunk 1 above)
Comment (gaoyangxiaozhu) [reply gluten-core/src/main/scala/org/apache/gluten/execution/SampleExecTransformer.scala:35]: sure, will do

Comment (gaoyangxiaozhu) [reply gluten-core/src/main/scala/org/apache/gluten/execution/SampleExecTransformer.scala:48]: done

**Review by rui-mo — APPROVED** [review]
Thanks.

**Comment by gaoyangxiaozhu** [comment]
@zhli1142015 help merge

**Pull request merged**